diff --git a/Makefile b/Makefile
index 765bfb2a06..df00ac8bbb 100644
--- a/Makefile
+++ b/Makefile
@@ -328,5 +328,5 @@ CXXFLAGS += -std=c++11
 else
 CXXFLAGS += -std=c++0x
 endif
-CPPFLAGS += -g -Wall -Wextra -Werror -Wno-long-long -Wno-unused-parameter -DOSATOMIC_USE_INLINED=1
+CPPFLAGS += -g -Wall -Wextra -Wno-long-long -Wno-unused-parameter -DOSATOMIC_USE_INLINED=1
 LDFLAGS += -g

 CPPFLAGS += $(CPPFLAGS_$(CONFIG))
